PhotoMap/Features/Profile/Coordinators/ProfileCoordinator.swift

```diff
@@ -0,0 +1,81 @@
+    private(set) var childCoordinators = [Coordinator]()
+    private(set) var navigationController = UINavigationController()
+    private let authService: AuthUserServiceType
+    private let fileManager: FileManagerServiceType
+    private let diContainer: DIContainerType
+    private let cancelBag = CancelBag()
+    
+    private(set) var logoutButtonSubject = PassthroughSubject<String, Never>()
+    private(set) var showErrorSubject = PassthroughSubject<GeneralErrorType, Never>()
+    
+    // MARK: - Lifecycle
+    init(diContainer: DIContainerType) {
+        self.diContainer = diContainer
+        self.authService = diContainer.resolve()
+        self.fileManager = diContainer.resolve()
+        bind()
+    }
+    
+    private func bind() {
+        logoutButtonSubject.sink(receiveValue: { [weak self] email in
+            self?.showLogoutAlert(with: email)
+        })
+        .store(in: cancelBag)
+        
+        showErrorSubject.sink(receiveValue: { [weak self] error in
+            self?.showErrorAlert(error: error)
+        })
+        .store(in: cancelBag)
+    }
+    
+    @discardableResult
+    func start() -> UIViewController {
+        let viewModel = ProfileViewModel(coordinator: self, diContainer: diContainer)
+        let profileVC = ProfileViewController.newInstanse(viewModel: viewModel)
+        profileVC.tabBarItem.title = L10n.Main.TabBar.Profile.title
+        profileVC.tabBarItem.image = UIImage(systemName: "person.crop.circle")
+        navigationController.pushViewController(profileVC, animated: true)
+        
+        return navigationController
+    }
+    
+    // MARK: - Helpers
+    func showLogoutAlert(with email: String) {
+        let alert = UIAlertController(title: L10n.Profile.Alert.Logout.title(email),
+                                      message: nil, preferredStyle: .alert)
+        let cancelAction = UIAlertAction(title: L10n.Profile.Alert.Logout.Action.cancel, style: .default)
+        let logoutAction = UIAlertAction(title: L10n.Logout.Button.Title.logOut, style: .destructive) { [weak self] _ in
+            self?.navigationController.dismiss(animated: true)
+            self?.logoutFromAccount()
+        }
+        alert.addAction(cancelAction)
+        alert.addAction(logoutAction)
+        navigationController.present(alert, animated: true)
+    }
+    
+    private func logoutFromAccount() {
+        guard let scene = UIApplication.shared.connectedScenes.first else { return }
+        guard let sceneDelegate = scene.delegate as? SceneDelegate else { return }
+        guard let window = sceneDelegate.window else { return }
+        guard let appCoordinator = sceneDelegate.appCoordinator else { return }
+        // swiftlint:disable line_length
+        UIView.transition(with: window, duration: 0.5, options: .transitionFlipFromLeft, animations: nil) { [weak self] _ in
+        // swiftlint:enable line_length
+            self?.authService.logOut()
+            self?.fileManager.clearCache()
```

Comment:
Coordinator should only contain navigation logic. Move it outside (e.g. in ViewModel)